sympy/core/logic.py

```diff
@@ -162,6 +162,7 @@ def fuzzy_not(v):
     True
 
     """
+    v = fuzzy_bool(v)
```

Comment:
I did this because, in case, if `v` cannot be reduced to `bool` type, then `not v` will not work, hence, converting it to `fuzzy_bool(v)` will make the the code below it work correctly. 
I will remove this changes, if not needed.

Reply:
Is this needed for some reason?

I think fuzzy_not should only be used with a fuzzy bool so the input should already be only True, False or None.